### PR TITLE
 TkDQM: PixelPhase1 Summary plots

### PR DESCRIPTION
**Pull request by sayanchatterjee38**
#### Title: 

TkDQM: PixelPhase1 Summary plots 

#### PR description:

The aim of this PR is to Implement a 3 state (GOOD, WARNING, ERROR) Summary Maps for PixelPhase1 in ONLINE DQM.  Pixel online client tested locally over RAW data. 

#### Links of the last presentation in TkDQM bi-weekly meeting:

https://indico.cern.ch/event/1000820/#b-407796-tracker-dqm 

#### PR validation:

Checks are done with workflow 136.892 and no changes are observed.

#### If this PR is a backport please specify the original PR and why you need to backport that PR:

NA

### CODE REVIEW

**Comment by sroychow** [comment]
@jfernan2 can you comment on the unit test failure of ```TestDQMServicesDemo```?

**Comment by perrotta** [comment]
please test with #35921

**Comment by perrotta** [comment]
please abort

**Comment by perrotta** [comment]
please test with #35921,#35766

**Comment by jfernan2** [comment]
@sayanchatterjee38  the PR looks OK but I am confused by your PR description since you say:

> Checks are done with workflow 136.892 and no changes are observed.

but the plot changes (for better), e.g.:  https://tinyurl.com/yeyzhe55

**Comment by sayanchatterjee38** [comment]
> but the plot changes (for better), e.g.: https://tinyurl.com/yeyzhe55

Hi @jfernan2, the definition of the plot you mentioned is :
https://github.com/cms-sw/cmssw/blob/5fb06adb43f84e24c0e2c29840cd61f4b78cf0c5/DQM/SiPixelPhase1Summary/src/SiPixelPhase1Summary.cc#L142

Here the differences are coming due to these two lines :
https://github.com/cms-sw/cmssw/blob/5fb06adb43f84e24c0e2c29840cd61f4b78cf0c5/DQM/SiPixelPhase1Summary/src/SiPixelPhase1Summary.cc#L286-L287 

Here when we said we expect no changes - it means the changes in the bins which are already populated.

**Comment by jfernan2** [comment]
+1

File: DQM/SiPixelPhase1Summary/src/SiPixelPhase1Summary.cc [hunk 1]
```diff
@@ -325,20 +328,40 @@ void SiPixelPhase1Summary::fillSummaries(DQMStore::IBooker& iBooker, DQMStore::I
...
+          else if (deadROCSummary->getBinContent(i + 1, j + 1) > deadRocWarnThresholds_[i * 4 + j] &&
+                   deadROCSummary->getBinContent(i + 1, j + 1) < deadRocThresholds_[i * 4 + j])
+            summaryMap_["Grand"]->setBinContent(i + 1, j + 1, 0.8);
+
+          else
+            summaryMap_["Grand"]->setBinContent(i + 1, j + 1, 0);
+
+          sumOfNonNegBins += summaryMap_["Grand"]->getBinContent(i + 1, j + 1);
+        }
+        if (i == 1 && j > 1) {
```
Comment (perrotta) [inline DQM/SiPixelPhase1Summary/src/SiPixelPhase1Summary.cc:346]: I miss why the very same code has to be repeated twice (L331-345 vs L346-360)

Comment (sayanchatterjee38) [reply DQM/SiPixelPhase1Summary/src/SiPixelPhase1Summary.cc:346]: Hi @perrotta, My apologies! I think somehow these lines had been added twice. Is there any way to remove L346-360 from here or do I have to push it again?

Comment (perrotta) [reply DQM/SiPixelPhase1Summary/src/SiPixelPhase1Summary.cc:346]: Thank you @sayanchatterjee38 
I think that pushing a new commit is the easiest way to proceed.
Only if you are at ease with it, you could further squeeze all commits into a single one afterwards

**Comment by sroychow** [comment]
@cmsbuild please test

**Comment by jfernan2** [comment]
+1

**Comment by perrotta** [comment]
+1

**Pull request merged**